nservicebus/azure-service-bus/transaction-support.md

```diff
@@ -7,37 +7,66 @@ tags:
 
-### Versions 6 and above
 
-Azure Service Bus Transport supports `SendAtomicWithReceive`, `ReceiveOnly` and `None` levels.
+#### Transport transaction - Receive Only
 
+The `ReceiveOnly` guarantee is based on the Azure Service Bus Peek-Lock mechanism.
 
-#### Transport transaction - Sends atomic with Receive
+The message is not removed from the queue directly after receive, but it's hidden for, by default , 30 seconds. That prevents other instances from picking it up. If the receiver fails to process the message withing that timeframe or explicitly abandons the message, then the message will become visible again. Other instances will be able to pick it up (effectively works like a rollback).
 
-Note: `SendAtomicWithReceive` level is supported only when destination and receive queues are in the same namespace.
 
-The `SendAtomicWithReceive` guarantee is achieved by using `ViaEntityPath` property on outbound messages. It's value is set to the receiving queue.
+#### Unreliable (Transactions Disabled)
 
-If the `ViaEntityPath` is not empty, then messages will be added to the receive queue. The messages will be forwarded to their actual destination (inside the broker) only when the complete operation is called on the received brokered message. The message won't be forwarded if the lock duration limit is exceeded (30 seconds by default) or if the message is explicitly abandoned.
+When transactions are disabled then NServiceBus the transport uses the [ASB's ReceiveAndDelete mode](https://msdn.microsoft.com/en-us/library/microsoft.servicebus.messaging.receivemode.aspx).
 
+The message is deleted from the queue directly after receive operation completes, before it is processed.
 
-#### Transport transaction - Receive Only
+This mode is highly unrecommended!
 
-The `ReceiveOnly` guarantee is based on the Azure Service Bus Peek-Lock mechanism.
 
-The message is not removed from the queue directly after receive, but it's hidden by default for 30 seconds. That prevents other instances from picking it up. If the receiver fails to process the message withing that timeframe or explicitly abandons the message, then the message will become visible again. Other instances will be able to pick it up.
+## Understanding internal transactions and delivery guarantees
 
+### Versions 6 and below
 
-#### Unreliable (Transactions Disabled)
+NServiceBus relies on transaction scopes to control how it behaves when it comes to transactions, message dispatching and commit/rollback guarantees. The architecture is schematically represented like this:
 
-When transactions are disabled then NServiceBus uses the [ASB's ReceiveAndDelete mode](https://msdn.microsoft.com/en-us/library/microsoft.servicebus.messaging.receivemode.aspx).
+[TODO: insert picture 1]
```

Comment:
Well done with the pic. First I thought it might be confusing that we have multiple boxes called pipeline. Then I realized this level of abstraction is helpful for this overview picture